tests/js/server/aql/aql-index-join-noncluster.js

```diff
@@ -0,0 +1,201 @@
+    }
+    assertNotEqual(planNodes.indexOf("JoinNode"), -1);
+
+    const result = AQL_EXECUTE(query, null, opts);
+    return result.json;
+  };
+
+  const databaseName = "IndexJoinDB";
+
+  return {
+    setUp: function () {
+      db._createDatabase(databaseName);
+      db._useDatabase(databaseName);
+    },
+
+    tearDown: function () {
+      db._useDatabase("_system");
+      db._dropDatabase(databaseName);
+    },
+
+    testEvenOdd: function () {
+      const A = fillCollection("A", singleAttributeGenerator(1000, "x", x => 2 * x + 1));
+      A.ensureIndex({type: "persistent", fields: ["x"]});
+      const B = fillCollection("B", singleAttributeGenerator(1000, "x", x => 2 * x));
+      B.ensureIndex({type: "persistent", fields: ["x"]});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 0);
+    },
+
+    testEveryOtherMatch: function () {
+      const A = fillCollection("A", singleAttributeGenerator(1000, "x", x => x));
+      A.ensureIndex({type: "persistent", fields: ["x"]});
+      const B = fillCollection("B", singleAttributeGenerator(500, "x", x => 2 * x));
+      B.ensureIndex({type: "persistent", fields: ["x"]});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 500);
+      for (const [a, b] of result) {
+        assertEqual(a.x, b.x);
+      }
+    },
+
+    testEveryMultipleMatches: function () {
+      const A = fillCollection("A", singleAttributeGenerator(100, "x", x => x));
+      A.ensureIndex({type: "persistent", fields: ["x"]});
+      const B = fillCollection("B", singleAttributeGenerator(1000, "x", x => x % 100));
+      B.ensureIndex({type: "persistent", fields: ["x"]});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 1000);
+      for (const [a, b] of result) {
+        assertEqual(a.x % 100, b.x);
+      }
+    },
+
+    testFullProduct: function () {
+      const A = fillCollection("A", singleAttributeGenerator(10, "x", x => 0));
+      A.ensureIndex({type: "persistent", fields: ["x"]});
+      const B = fillCollection("B", singleAttributeGenerator(10, "x", x => 0));
+      B.ensureIndex({type: "persistent", fields: ["x"]});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 100);
+      for (const [a, b] of result) {
+        assertEqual(a.x, b.x);
+      }
+    },
+/*
```

Comment:
Is this test intentionally commented out? Can it be activated for this PR already?

Reply:
The problem here is that the optimizer somehow does not convert the second for + sort into an index scan. I haven't looked into details but it appears to be a more general problem.